### PR TITLE
Add number of occupations next to states

### PR DESCRIPTION
**Pull request by jenlu33**
Asana ticket: [https://app.asana.com/0/1203289004376659/1204799581648913/f](https://app.asana.com/0/1203289004376659/1204799581648913/f)

Screenshot 2023-06-21 at 1 24 19 PM

### CODE REVIEW

File: app/controllers/states_controller.rb [hunk 1]
```diff
@@ -2,5 +2,6 @@ class StatesController < ApplicationController
   def index
     state_info = State.order(:name).pluck(:name, :id)
     @names_by_letter = state_info.group_by { |state| state[0][0] }
+    @occupation_standards_count_by_state = OccupationStandard.joins(registration_agency: :state).group("states.id").count
```
Comment (littleforest) [inline app/controllers/states_controller.rb:5]: @jenlu33 let's make this count a separate method in the model, and then you can add some unit tests for it.

Comment (littleforest) [reply app/controllers/states_controller.rb:5]: Maybe we could also beef up the index request spec making sure there are at least 2 of the same states to count so Bullet can help catch any query inefficiencies.

File: app/views/states/index.html.erb [hunk 2]
```diff
@@ -37,8 +37,9 @@
                 </div>
                     <div class="break-inside-avoid-column px-4 flex-grow">
                     <% names.each do |name, id| %>
-                        <%= link_to occupation_standards_path(state_id: id), class: "hover:font-bold" do %>
+                        <%= link_to occupation_standards_path(state_id: id), class: "hover:font-bold flex space-x-2" do %>
                             <span class="block"><%= name %></span>
+                            <span class="text-gray-500"><%= @occupation_standards_count_by_state[id] || "" %></span>
```
Comment (littleforest) [inline app/views/states/index.html.erb:42]: Were there any designs for this? I feel like it's a little hard to see the count, or even to know what it is for...

File: spec/system/states/index_spec.rb [hunk 3]
```diff
@@ -1,22 +1,22 @@
 require "rails_helper"
 
 RSpec.describe "states/index" do
-  it "lists states with link to search by state" do
+  it "lists states with link to search by state and shows occupatoin standards count" do
```
Comment (sarahraqueld) [inline spec/system/states/index_spec.rb:4]:
```suggestion
  it "lists states with link to search by state and shows occupation standards count" do
```

File: app/views/states/index.html.erb [hunk 4]
```diff
@@ -37,9 +37,9 @@
                 </div>
                     <div class="break-inside-avoid-column px-4 flex-grow">
                     <% names.each do |name, id| %>
-                        <%= link_to occupation_standards_path(state_id: id), class: "hover:font-bold" do %>
+                        <%= link_to occupation_standards_path(state_id: id), class: "hover:font-bold flex items-center space-x-1" do %>
                             <span class="block"><%= name %></span>
-                        <% end %>
+                            <span class="text-xs text-gray-500"><%= State.find(id).occupation_standards_count %></span>                        <% end %>
```
Comment (sarahraqueld) [inline app/views/states/index.html.erb:42]: Was this extra whitespace added intentionally? 

```suggestion
                 <span class="text-xs text-gray-500"><%= State.find(id).occupation_standards_count %></span>                     <% end %>
```

**Review by sarahraqueld — APPROVED** [review]
Looks good!

**Pull request merged**